nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/ListInstancesCommand.java

```diff
@@ -214,32 +214,32 @@ private boolean notDas(String name) {
         return !SystemPropertyConstants.DAS_SERVER_NAME.equals(name);
     }
 
-    private void yesStatus(List<Server> serverList, int timeoutInMsec, Logger logger) {
+    private void yesStatus(final List<Server> serverList, final int timeoutInMsec, final Logger logger) {
         // Gather a list of InstanceInfo -- one per instance in domain.xml
-        RemoteInstanceCommandHelper helper = new RemoteInstanceCommandHelper(habitat);
+        final RemoteInstanceCommandHelper helper = new RemoteInstanceCommandHelper(habitat);
```

Comment:
final

Reply:
> Please remove `final` in your changes. That just adds clutter to the code and doesn't really do anything

I have to disagree with following reasons:

1. it should help to distinguish which variables are variable and
2. final variables does the code flow pretty staight forward. I mean if variable is final, there are no side effects and I can concentrate to pitfals of reused variables (wich may be really nasty). For example trying to find where regis variable comes consumes a lot of brain power (which have to be gathered from chocolate or coffe): https://github.com/payara/Payara/blob/75cc47823f5f9d6f401fa84fcdbaef8bbd55a310/appserver/web/web-core/src/main/java/org/apache/catalina/core/StandardContext.java#L3624 
3. final variables reduces scope where variable might be set and improves learning curve of code for contributors (by reducing possibilities where variable can be changed)
4. non final variables are like exclamation mark and acts like point of interest in case of hunting bugs
5. in old codebase final reveals scoping problems (

```
before (C89 declaration style):
String x = null;
while (...) {
  x = "some String";
  ...
}

after (Java style):
while (...) {
  final String x = "some String";
  ...
}
```

I know some time ago there was a discussion, but I do not know the final decision.

For reference there is PR in commons-lang3 with the similar idiom. https://github.com/apache/commons-lang/pull/687/commits/37c22318bc3e33c3df3a60afb5aac68659eaeadc

Another insteresting references
- https://softwareengineering.stackexchange.com/questions/98691/excessive-use-final-keyword-in-java
- https://stackoverflow.com/questions/137868/using-the-final-modifier-whenever-applicable-in-java

So if there is consensus on that the final is just garbage, I can discard the last commit.

Personally I would preffer to use final at least in methods that are longer than usual and contains if constructs. In case final keyword is not wanted in all possible places.

Reply:
Bu you are right in code review, it should be in different PR.